packages/react-components/react-utilities/src/hooks/useId.ts

```diff
@@ -34,10 +34,15 @@ export function useId(prefix: string = 'fui-', providedId?: string): string {
 
   // Checking if useId is available on React, if it is, we use it to generate the id. String concatenation is used to
   // prevent bundlers from complaining with older versions of React.
-  const _useId: () => string | undefined = (React as never)['use' + 'Id'];
+  const _useId = (React as never)['use' + 'Id'] as (() => string) | undefined;
```

Comment:
Adding a type `_useId: (() => string) | undefined` will not work as it will resolve as `never`, to get around that we have to type cast it to that type instead.